TTS/tts/models/base_tts.py

```diff
@@ -12,11 +12,12 @@
 from TTS.model import BaseModel
 from TTS.tts.configs.shared_configs import CharactersConfig
 from TTS.tts.datasets.dataset import TTSDataset
-from TTS.tts.utils.languages import LanguageManager, get_language_weighted_sampler
-from TTS.tts.utils.speakers import SpeakerManager, get_speaker_manager, get_speaker_weighted_sampler
+from TTS.tts.utils.languages import LanguageManager, get_language_balancer_weights
```

Comment:
Why don't you compute sampler weights in the Managers? They already have all the info necessary for computing the weights.

Reply:
Indeed, but maybe we can use it in the speaker encoder/vocoder/deep fake detector. In this case, maybe its better to keep it as functions (because in these other models Language/Speaker manager will be not used). But if you like I can move to Language/Speaker manager and in the future, if necessary we can just copy the function.